executor.go

```diff
@@ -277,6 +277,10 @@ func (e *Executor) provideEventLogger(t *Task) *Logger {
 
 // runPass kicks off tasks that are in an executable state.
 func (e *Executor) runPass() {
+	if e.ctx.Err() != nil {
```

Comment:
Should this check more explicitly if `e.ctx.Err() == context.Cancelled`? Is context cancelled the only error that could be on the ctx?

Reply:
context's error is always canceled or deadline exceeded - in both cases, the executor is done here